SDL/Quintuplet.cu

```diff
@@ -428,6 +428,117 @@ __device__ bool SDL::runQuintupletDefaultAlgo(struct SDL::modules& modulesInGPU,
     pass = pass and tempPass;
     if(not pass) return pass;
 
+    short layer2_adjustment = 0;
+    if (modulesInGPU.layers[lowerModuleIndex1] == 1)
+    {
+        layer2_adjustment = 1; // get upper segment to be in second layer
+    }
+    bool is_endcap1 = (modulesInGPU.subdets[lowerModuleIndex1] == 4);
+    bool is_endcap2 = (modulesInGPU.subdets[lowerModuleIndex2] == 4);
+    bool is_endcap3 = (modulesInGPU.subdets[lowerModuleIndex3] == 4);
+    bool is_endcap4 = (modulesInGPU.subdets[lowerModuleIndex4] == 4);
+    bool is_endcap5 = (modulesInGPU.subdets[lowerModuleIndex5] == 4);
+    float x[38] = { 
+        log10(2.99792458e-3f*3.8f*innerRadius),                       // inner t3_pt
+        mdsInGPU.anchorEta[firstMDIndex],                             // inner (hit 1) t3_0_eta
+        mdsInGPU.anchorPhi[firstMDIndex],                             // inner (hit 1) t3_0_phi
+        mdsInGPU.anchorZ[firstMDIndex],                               // inner (hit 1) t3_0_z
+        sqrtf(x1*x1 + y1*y1),                                         // inner (hit 1) t3_0_r
```

Comment:
Maybe this is not for this PR but would it make sense to use r^2, instead of r, to avoid extra, costly operations?

The same comment applies to more instances below.

Reply:
If we used r^2 instead, I think we'd anyway need to normalize the value like we do for e.g. the pTs. In any case, I expect the `sqrt` call is a small cost compared to the for loops, no?

Reply:
Why normalize the value if it is `r^2` and not when it is `r`?

Maybe I have missed the point but my understanding was that you are using `r` as an input variable in the training (and inference), is that right? If that's the case, what I was proposing was to use `r^2` instead of `r` in the training, thinking that it would have similar discriminatory power but saving us the `sqrt` downstream.

Reply:
Sorry, to be clear, we need to normalize DNN input features such that they are all O(1). For example, we take the log10 of the T3 pTs and T5 pT. Thus, if we were to give the DNN r^2 instead of r, it would be O(100), which we would need to normalize like we do for the pTs. Does that make sense?

Reply:
I see. But then r = sqrt( O(100) ) = O(10), which is still one order of magnitude larger than O(1). For my education, is this fine? Where do we set the limit for normalization? Is it an empirical limit by checking if the training does well or something else?

Reply:
Yes, at least for what we're doing right now. I've seen for other problems that normalizing anything that's more flat (e.g. phi) such that the values are made to be between 0 and 1 also helps. However, since the DNN is converging, we can say "taking the log of the pTs is enough for now."